bundles/org.openhab.binding.dbquery/src/main/java/org/openhab/binding/dbquery/action/DBQueryActions.java

```diff
@@ -0,0 +1,125 @@
+import org.openhab.binding.dbquery.internal.domain.QueryResult;
+import org.openhab.binding.dbquery.internal.domain.ResultRow;
+import org.openhab.binding.dbquery.internal.error.UnnexpectedCondition;
+import org.openhab.core.automation.annotation.ActionInput;
+import org.openhab.core.automation.annotation.RuleAction;
+import org.openhab.core.thing.binding.ThingActions;
+import org.openhab.core.thing.binding.ThingActionsScope;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Joan Pujol - Initial contribution
+ */
+@ThingActionsScope(name = "dbquery")
+@NonNullByDefault
+public class DBQueryActions implements IDBQueryActions, ThingActions {
+    private final Logger logger = LoggerFactory.getLogger(DBQueryActions.class);
+
+    private @Nullable QueryHandler queryHandler;
+    private @Nullable DatabaseBridgeHandler databaseBridgeHandler;
+
+    @Override
+    public void activate() {
+        logger.trace("activate actions");
+    }
+
+    @Override
+    public void deactivate() {
+        logger.trace("deactivate actions");
+    }
+
+    @Override
+    @RuleAction(label = "Execute query", description = "Execute query synchronously (use with care)")
+    public ActionQueryResult executeQuery(String query, Map<String, @Nullable Object> parameters,
+            int timeoutInSeconds) {
+        logger.debug("executeQuery from action {} params={}", query, parameters);
+        var currentDatabaseBridgeHandler = databaseBridgeHandler;
+        if (currentDatabaseBridgeHandler != null) {
+            QueryResult queryResult = new ExecuteNonConfiguredQuery(currentDatabaseBridgeHandler.getDatabase())
+                    .executeSynchronously(query, parameters, Duration.ofSeconds(timeoutInSeconds));
+            logger.debug("executeQuery from action result {}", queryResult);
+            return queryResult2ActionQueryResult(queryResult);
+        } else {
+            logger.warn("Execute queried ignored as databaseBridgeHandler is null");
+            return new ActionQueryResult(false, null);
+        }
+    }
+
+    private ActionQueryResult queryResult2ActionQueryResult(QueryResult queryResult) {
+        return new ActionQueryResult(queryResult.isCorrect(),
+                queryResult.getData().stream().map(DBQueryActions::resultRow2Map).collect(Collectors.toList()));
+    }
+
+    private static Map<String, @Nullable Object> resultRow2Map(ResultRow resultRow) {
+        Map<String, @Nullable Object> map = new HashMap<>();
+        for (String column : resultRow.getColumnNames()) {
+            map.put(column, resultRow.getValue(column));
+        }
+        return map;
+    }
+
+    @Override
+    @RuleAction(label = "Set query parameters", description = "Set query parameters for a query")
+    public void setQueryParameters(@ActionInput(name = "parameters") Map<String, @Nullable Object> parameters) {
+        logger.info("setQueryParameters {}", parameters);
```

Comment:
info logging should not be used very often, suggest this should be debug or trace?

https://www.openhab.org/docs/developer/guidelines.html#f-logging